spatialmath/geo_obstacle.go

```diff
@@ -110,3 +106,35 @@ func GeoObstaclesFromConfig(config *GeoObstacleConfig) ([]*GeoObstacle, error) {
 	}
 	return gobs, nil
 }
+
+// GetCartesianDistance calculates the great circle distance between p and q.
+func GetCartesianDistance(p, q *geo.Point) (float64, float64) {
```

Comment:
I don't understand why this exists when we can just call `p.GreatCircleDistance(q)`?

Reply:
I'm getting tripped up with why we need to reference this to (0, 0) but I think this makes sense and is fine as an MVP.  In my head I'm imagining a world where we rewrite the Pose interface such that we can support GeoPoses too, but this can come later

Reply:
I see what you are saying, lets get some opinions! @tessavitabile @biotinker

Reply:
I think this won't be accurate for long distances, since we're trying to map lat and long onto a grid, but they are not a grid. I think for now we could just use lat and long directly, and just impose a limit on how far we're willing to travel using MoveOnGlobe. But I don't feel that strongly.

Reply:
I like the idea of imposing a limit on single call distances. There are a number of places within our planning system where default values are selected based on the assumption that the motion system is moving in a region approximately human sized; on the mm to tens of m scale.

We will see a number of emergent issues such as floating point precision drop-out if we use the current system to move many kilometers at once without adjustment.

@nfranczak could you add a ticket/TODO to ensure configurability of the motion system for very large distances, so that this is tracked?

Reply:
https://viam.atlassian.net/browse/RSDK-3559